Mathlib/RingTheory/TensorProduct/Pi.lean

```diff
@@ -47,12 +48,44 @@ noncomputable def piRightHom : A ⊗[R] (∀ i, B i) →ₐ[S] ∀ i, A ⊗[R] B
 variable [Fintype ι] [DecidableEq ι]
 
 /-- Tensor product of rings commutes with finite products on the right. -/
-noncomputable def Algebra.TensorProduct.piRight :
-    A ⊗[R] (∀ i, B i) ≃ₐ[S] ∀ i, A ⊗[R] B i :=
+noncomputable def piRight : A ⊗[R] (∀ i, B i) ≃ₐ[S] ∀ i, A ⊗[R] B i :=
   AlgEquiv.ofLinearEquiv (_root_.TensorProduct.piRight R S A B) (by simp) (by simp)
 
 @[simp]
-lemma Algebra.TensorProduct.piRight_tmul (x : A) (f : ∀ i, B i) :
+lemma piRight_tmul (x : A) (f : ∀ i, B i) :
     piRight R S A B (x ⊗ₜ f) = (fun j ↦ x ⊗ₜ f j) := rfl
 
+variable (ι) in
+/-- Variant of `Algebra.TensorProduct.piRight` with constant factors. -/
+noncomputable def piScalarRight : A ⊗[R] (ι → R) ≃ₐ[S] ι → A :=
+  (piRight R S A (fun _ : ι ↦ R)).trans <|
+    AlgEquiv.piCongrRight (fun _ ↦ Algebra.TensorProduct.rid R S A)
+
+@[simp]
+lemma piScalarRight_tmul (x : A) (y : ι → R) :
+    piScalarRight R S A ι (x ⊗ₜ y) = fun i ↦ y i • x :=
+  rfl
```

Comment:
I think this might be useful but `piScalarRight_tmul_apply` should be the simp lemma instead.